project/src/js/socket.ts

```diff
@@ -67,10 +71,15 @@ const defaultHandlers: MessageHandlers = {
   }
 };
 
-function handleFollowingOnline(data: Array<string>) {
-  const curList = friendsApi.list();
-  friendsApi.set(data);
-  if (xor(curList, data).length > 0) {
+function handleFollowingOnline(data: Array<string>, payload: any) {
+  const oldFriendList = friendsApi.list().slice();
```

Comment:
This should actually use lodash's 'deepCopy'.

A deep copy is necessary as we're comparing the old version of the friendsApi.list() array which may have been mutated in place when a user played / stopped playing.

Shall fix tonight!